src/backend/gl/src/factory.rs

```diff
@@ -392,6 +392,13 @@ impl f::Factory<R> for Factory {
         let mut inputs = [None; d::MAX_VERTEX_ATTRIBUTES];
         for i in 0 .. d::MAX_VERTEX_ATTRIBUTES {
             inputs[i] = desc.attributes[i].map(|at| BufferElement {
+                // Note: this `Option::unwrap` call fails the emscripten (sdk-1.38.16-64bit) build. Inlining the call seems to work around the issue. (See #2721)
+                #[cfg(target_os = "emscripten")]
+                desc: match desc.vertex_buffers[at.0 as usize] {
```

Comment:
should we just do this for all targets unconditionally? (with the comment preserved, of course)

Reply:
I'm unsure. I'm personally a fan of keeping hacks as closely scope to their specific condition, but I think the style is a decision of the project, so I'll defer to you. I'm happy to make the change, if you'd like.